### PR TITLE
Tidy: Report invalid listed ignored directories and files

### PR DESCRIPTION
**Pull request by UK992**
r? @Wafflespeanut

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/15035)

### CODE REVIEW

File: python/tidy/servo_tidy/tidy.py [hunk 1]
```diff
@@ -827,6 +830,17 @@ def check_config_file(config_file, print_text=True):
     if print_text:
         print '\rChecking the config file...'
 
+    config_content = toml.loads(conf_file)
+    exclude = config_content.get("ignore", {})
+
+    # Check for invalid listed ignored directories
+    exclude_dirs = exclude.get("directories", [])
+    skip_dirs = ["./target", "./tests"]
```
Comment (Wafflespeanut) [inline python/tidy/servo_tidy/tidy.py]: We can check for `"target"` but we can safely remove the dirs (from the config) that don't exist in `"tests"`. It's old code :smile:

File: python/tidy/servo_tidy/tidy.py [hunk 2]
```diff
@@ -841,6 +855,22 @@ def check_config_file(config_file, print_text=True):
             current_table = table_name
             continue
 
+        # Print invalid listed ignored directories
+        if current_table == "ignore" and invalid_dirs:
+            for d in invalid_dirs:
+                if any(fd in line for fd in ["'{}'".format(d), '"{}"'.format(d)]):
```
Comment (Wafflespeanut) [inline python/tidy/servo_tidy/tidy.py]: Couldn't we have something like this?

``` python
if line.strip().strip('\'"') == d:
```

File: python/tidy/servo_tidy/tidy.py [hunk 3]
```diff
@@ -841,6 +855,22 @@ def check_config_file(config_file, print_text=True):
...
+                    yield config_file, idx + 1, "invalid listed ignored directory '%s'" % d
+                    invalid_dirs.remove(d)
+                    break
+
+        # Print invalid listed ignored files
+        if current_table == "ignore" and invalid_files:
+            for f in invalid_files:
+                if any(ff in line for ff in ["'{}'".format(f), '"{}"'.format(f)]):
```
Comment (Wafflespeanut) [inline python/tidy/servo_tidy/tidy.py]: ditto

**Review by ghost — COMMENTED** [review]
Just a few comments.

**Comment by Wafflespeanut** [comment]
r=me after addressing that.

@bors-servo delegate+

File: python/tidy/servo_tidy/tidy.py [hunk 4]
```diff
@@ -841,6 +855,22 @@ def check_config_file(config_file, print_text=True):
             current_table = table_name
             continue
 
+        # Print invalid listed ignored directories
+        if current_table == "ignore" and invalid_dirs:
+            for d in invalid_dirs:
+                if any(fd in line for fd in ["'{}'".format(d), '"{}"'.format(d)]):
+                    yield config_file, idx + 1, "invalid listed ignored directory '%s'" % d
```
Comment (Wafflespeanut) [inline python/tidy/servo_tidy/tidy.py]: Oh, and also let's have a more specific help message, something like, `"ignored directory %r doesn't exist" % d`

File: python/tidy/servo_tidy/tidy.py [hunk 5]
```diff
@@ -841,6 +855,22 @@ def check_config_file(config_file, print_text=True):
...
+                    invalid_dirs.remove(d)
+                    break
+
+        # Print invalid listed ignored files
+        if current_table == "ignore" and invalid_files:
+            for f in invalid_files:
+                if any(ff in line for ff in ["'{}'".format(f), '"{}"'.format(f)]):
+                    yield config_file, idx + 1, "invalid listed ignored file '%s'" % f
```
Comment (Wafflespeanut) [inline python/tidy/servo_tidy/tidy.py]: ditto on message

File: python/tidy/servo_tidy/tidy.py (hunk 1 above)
Comment (UK992) [inline python/tidy/servo_tidy/tidy.py]: what about `./tests/wpt/sync` and `./tests/wpt/sync_css` which does not exist, but they are in [css](https://github.com/servo/servo/blob/master/tests/wpt/config_css.ini#L9) and [wpt](https://github.com/servo/servo/blob/master/tests/wpt/config.ini#L9) config files, or is that obsolete?

File: python/tidy/servo_tidy/tidy.py (hunk 5 above)
Comment (UK992) [inline python/tidy/servo_tidy/tidy.py]: done

File: python/tidy/servo_tidy/tidy.py (hunk 4 above)
Comment (UK992) [inline python/tidy/servo_tidy/tidy.py]: done

File: python/tidy/servo_tidy/tidy.py (hunk 2 above)
Comment (UK992) [inline python/tidy/servo_tidy/tidy.py]: done

File: python/tidy/servo_tidy/tidy.py (hunk 3 above)
Comment (UK992) [inline python/tidy/servo_tidy/tidy.py]: done

File: python/tidy/servo_tidy/tidy.py (hunk 1 above)
Comment (Wafflespeanut) [inline python/tidy/servo_tidy/tidy.py]: Looks like that's been used for sync'ing the WPT stuff. We better leave it there :)

**Comment by Wafflespeanut** [comment]
@bors-servo r+

Thanks!

**Pull request merged**